cmd/gateway-azure.go

```diff
@@ -37,6 +40,7 @@ import (
 
 const globalAzureAPIVersion = "2016-05-31"
 const azureBlockSize = 100 * humanize.MiByte
+const metadataObjectTemplate = "minio.sys.tmp/multipart/v1/%s/meta.json"
```

Comment:
instead you can have

```
func getAzureMultipartMetadataPath(bucket, object, uploadID string) string {
    id := sha256(bucket+object) + "." + uploadID
    return fmt.Sprintf("minio.sys.tmp/multipart/v1/%s/meta.json", id)
}
```
since uploadID is just 8 bytes having sha256 of bucket and object will make this more hash collision resistant

Reply:
Encoding bucket and object name would prevent recovery case using upload ID.   However this does not prevent the problem completely in situation like in another instance of gateway gets previously generated upload ID for same bucket/object would lead to meta data loss.

To avoid these problem (to some extent), I added upload id existent check in new multipart upload call.

> you can name the file azure.json as we have gcs.json for gcs

Done

Reply:
> Encoding bucket and object name would prevent recovery case using upload ID.

it can still be done as the uploadID is part of the metadataPath (metadataPath also contains sha256 which is additional info)

> However this does not prevent the problem completely in situation like in another instance of gateway gets previously generated upload ID for same bucket/object would lead to meta data loss.

That is right, what I pointed out is that we can make hash collision more resistant (than what it is currently) by including sha256 of `bucket/object`, 

> To avoid these problem (to some extent), I added upload id existent check in new multipart upload call.

OK, if this is done then it is good enough.